packages/@ourworldindata/grapher/src/stackedCharts/AbstractStackedChart.tsx

```diff
@@ -84,6 +84,31 @@ export class AbstractStackedChart
         return table
     }
 
+    transformTableForSelection(table: OwidTable): OwidTable {
+        // if entities with partial data are not plotted,
+        // make sure they don't show up in the entity selector
+        if (this.missingDataStrategy !== MissingDataStrategy.show) {
+            table = table
+                .replaceNonNumericCellsWithErrorValues(this.yColumnSlugs)
+                .dropRowsWithErrorValuesForAllColumns(this.yColumnSlugs)
+
+            if (this.shouldRunLinearInterpolation) {
+                this.yColumnSlugs.forEach((slug) => {
+                    table = table.interpolateColumnLinearly(slug)
+                })
+            }
+
+            const groupedByEntity = table
+                .groupBy("entityName")
+                .map((t: OwidTable) =>
+                    t.dropRowsWithErrorValuesForAnyColumn(this.yColumnSlugs)
+                )
+            table = groupedByEntity[0].concat(groupedByEntity.slice(1))
```

Comment:
I think we should also have the guard against `groupedByEntity.length === 0` here.

Reply:
lol yes, sorry for making you spell this out twice 🤪